atox/src/main/kotlin/ToxService.kt

```diff
@@ -90,23 +93,27 @@ class ToxService : LifecycleService() {
         createNotificationChannel()
         startForeground(NOTIFICATION_ID, notificationFor(connectionStatus))
 
-        userRepository.get(tox.publicKey.string())
-            .filter { user: User? -> user != null }.asLiveData().observe(this) { user ->
-                if (user.connectionStatus == connectionStatus) return@observe
-                connectionStatus = user.connectionStatus
-                notifier.notify(NOTIFICATION_ID, notificationFor(connectionStatus))
-                if (connectionStatus == ConnectionStatus.None) {
-                    Log.i(TAG, "Gone offline, scheduling bootstrap")
-                    bootstrapTimer.schedule(BOOTSTRAP_INTERVAL_MS, BOOTSTRAP_INTERVAL_MS) {
-                        Log.i(TAG, "Been offline for too long, bootstrapping")
-                        tox.isBootstrapNeeded = true
+        lifecycleScope.launch(Dispatchers.Default) {
+            userRepository.get(tox.publicKey.string())
+                .filterNotNull()
+                .filter { it.connectionStatus != connectionStatus }
+                .flowWithLifecycle(lifecycle)
```

Comment:
Do we really need this? From what I understood it's for consuming flows from the UI layer when an activity for example would go to the back stack and you should no longer need to consume it. But this one runs in the foreground anyway and it runs indefinitely.

Reply:
The service provides a lifecycle, so it's more "obviously correct" to use it since we want this to only run when the service is started.

Afaik, it goes `CREATE` -> `START` -> `STOP` -> `DESTROY`, and flowWithLifecycle is only started while the lifecycle is in `START` or higher by default, so it does something, and the official docs seem to always use it. :P

That said, this should also work without it, even if it starts running during `CREATE` and not during `START`. :x

Reply:
OK. Seems fine then.